src/main.rs

```diff
@@ -490,22 +490,28 @@ fn main_try() -> Result<()> {
         let defmt_state = if defmt_enable {
             let elf = fs::read(path).unwrap();
             let table = defmt_decoder::Table::parse(&elf)?;
-
-            let locs = {
-                let table = table.as_ref().unwrap();
-                let locs = table.get_locations(&elf)?;
-
-                if !table.is_empty() && locs.is_empty() {
-                    log::warn!("Insufficient DWARF info; compile your program with `debug = 2` to enable location info.");
-                    None
-                } else if table.indices().all(|idx| locs.contains_key(&(idx as u64))) {
-                    Some(locs)
-                } else {
-                    log::warn!("Location info is incomplete; it will be omitted from the output.");
-                    None
-                }
-            };
-            Some((table.unwrap(), locs))
+            if table.is_none() {
```

Comment:
You can make this an `if let Some(table) = table` and invert the logic (else block in if block (as it is of right now) and vice versa), then you can get rid of the `.unwrap()` on lone 498 and further occurences :)